Add query planning CPU time stat

weijiii opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #dev in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description
This PR aims to close #2579 

I think the thread in which planning begins does not need to be tracked when writing to `QueryStateTimer#planningCpuTime`. For DDL queries the whole state transitioning happens on the same thread. For normal queries `beginPlanning` and `beginStarting` are supposed to run on the same thread in a linear fashion, hence the first attempt to set `planningCpuTime` should yield correct result.

The following image shows the change on the UI. I suppose this should also be in `QueryCompletedEvent` so I will work on that and update this PR.

![Screenshot 2023-04-01 at 3 35 18 AM](https://user-images.githubusercontent.com/92953539/229281340-3acc911d-5140-4e07-b16b-86a1c752c99c.png)


(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

weijiii commented:
Seems a system connector test is failing. Will fix.

weijiii commented:
Added item in `trino-spi/pom.xml` because number of parameters changed for `QueryStatistics`

weijiii commented:
@sopel39 Hi, can you take a look? Thanks.

sopel39 reviewed (approved):
lgtm % I think you should extract mysql event listener changes to separate (follow-up) PR

arhimondr commented on plugin/trino-mysql-event-listener/src/main/java/io/trino/plugin/eventlistener/mysql/MysqlEventListener.java:
```diff
@@ -131,6 +131,7 @@ public void queryCompleted(QueryCompletedEvent event)
                 stats.getResourceWaitingTime().map(Duration::toMillis).orElse(0L),
                 stats.getAnalysisTime().map(Duration::toMillis).orElse(0L),
                 stats.getPlanningTime().map(Duration::toMillis).orElse(0L),
+                stats.getPlanningCpuTime().map(Duration::toMillis).orElse(0L),
```
This change is braking. Though the MySQL listener is not designed to be used in production as of today and is built for testing purposes only. We don't have to maintain strong compatibility.

sopel39 replied:
Thanks. In that case I thin MySQL listener should be out of scope of this PR

weijiii replied:
@sopel39 @arhimondr 
Thanks for the review. I'd like to finish this PR. I will
* Rebase w.r.t the latest commit
* Extract the changes on `trino-mysql-event-listener` to a separate PR

weijiii commented:
I've rebased and extracted the changes on `trino-mysql-event-listener` to https://github.com/trinodb/trino/pull/16841
The current test failures are due to `trino-mysql-event-listener` build. Please take a look. Thanks! 
cc: @sopel39 @arhimondr

sopel39 commented:
> The failures are caused by the missing parts for trino-mysql-event-listener

Could you make the build pass?

weijiii commented:
> > For all CI tasks to pass I will need to include the changes on mysql event listener.
> 
> I'm not sure why. You can have extra field in `QueryStatistics`, but not necessarily mysql changes as part of this PR. The UI change on it's own is very useful

Thanks let me try that.

weijiii commented:
@sopel39 Rebased and excluded `trino-mysql-event-listener` DAO

sopel39 commented on core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java:
```diff
@@ -683,6 +683,7 @@ private QueryStats getQueryStats(Optional<StageInfo> rootStage, List<StageInfo>
                 queryStateTimer.getAnalysisTime(),
                 queryStateTimer.getPlanningTime(),
                 queryStateTimer.getFinishingTime(),
+                queryStateTimer.getPlanningCpuTime(),
```
could you move it immediately below `getPlanningTime()` (here and in other places)?

weijiii replied:
Sure

sopel39 reviewed (approved):
lgtm % comment

sopel39 commented:
merged, thanks!
